### DISCUSSION TITLE
Failing tests

### DISCUSSION THREAD

**Discussion by arnabghosh3654** [eclipse-edc/Connector#884] [General]
While running ./gradlew clean build, I get he following failing tests:
DataPlaneHttpIntegrationTests > transfer_sinkTemporaryDropsConnection_noRetry() FAILED
    java.lang.AssertionError: Request not found exactly 2 times, expected:<{
      "body" : {
        "type" : "BINARY",
        "base64Bytes" : "MjE5NTcyNzMtN2U5Zi00ZDBmLWI2NmQtZGFjMDA2ZTIyMjA1"
      },
      "headers" : {
        "authorization" : [ "laborum" ],
        "content-type" : [ "application/octet-stream" ]
      },
      "method" : "POST",
      "path" : "/sample"
    }> but was:<{
      "body" : {
        "type" : "BINARY",
        "base64Bytes" : "MjE5NTcyNzMtN2U5Zi00ZDBmLWI2NmQtZGFjMDA2ZTIyMjA1",
        "contentType" : "application/octet-stream"
      },
      "headers" : {
        "authorization" : [ "laborum" ],
        "content-length" : [ "36" ],
        "Connection" : [ "Keep-Alive" ],
        "User-Agent" : [ "okhttp/4.9.3" ],
        "Host" : [ "localhost:42573" ],
        "Accept-Encoding" : [ "gzip" ],
        "Content-Type" : [ "application/octet-stream" ]
      },
      "keepAlive" : true,
      "method" : "POST",
      "path" : "/sample",
      "secure" : false
    }>
        at org.mockserver.client.MockServerClient.verify(MockServerClient.java:781)
        at org.eclipse.dataspaceconnector.dataplane.http.DataPlaneHttpIntegrationTests.transfer_sinkTemporaryDropsConnection_noRetry(DataPlaneHttpIntegrationTests.java:396)

6 tests completed, 1 failed

> Task :extensions:data-plane:integration-tests:test FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':extensions:ion:ion-client:test'.
> There were failing tests. See the report at: file:///home/arnab/newec/DataSpaceConnector-milestone-2/extensions/ion/ion-client/build/reports/tests/test/index.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':extensions:data-plane:integration-tests:test'.
> There were failing tests. See the report at: file:///home/arnab/newec/DataSpaceConnector-milestone-2/extensions/data-plane/integration-tests/build/reports/tests/test/index.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 13s
1029 actionable tasks: 1029 executed
---------------------------------------------------------
I am running behind a proxy and I have configured proxy settings in ~/gradle/gradle.properties . Any help is much appreciated.

**Comment by MoritzKeppler** [comment]
I have the same problem. @bscholtes1A @cpeeyush  any ideas?

**Reply by cpeeyush** [reply]
If you change this 1 does that solves the issue ?

Test : transfer_sinkTemporaryDropsConnection_noRetry

```
     
        httpSinkClientAndServer.verify(
                postRequest(body),
                VerificationTimes.exactly(2) <- change this to VerificationTimes.exactly(1) 
        );
```


Some background context:

originally when I was writing this test it took me sometime to figure out that why it's been called twice when we do not have retry policy for sink calls. After doing some research found out that it was okhttp client doing a retry internally, More details are here on PR discussion https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/758#discussion_r814666585

(Not sure why we do not use retry policy in HttpDataSink but we do use it in HttpDataSource)

**Reply by MoritzKeppler** [reply]
works with exactly(1) - maybe we can (at least as a temporary solution) change it to atLeast(1)?

**Reply by cpeeyush** [reply]
My original expectation was also that is should be exactly(1) because we do not have a retry policy for sink so it shouldn't be 2. But because of okhttp client behavior which I observed it was trying one more time and that's why kept it 2.

Not sure what has changed since then. If one works for you and on CI then sure you can change it because that is more appropriate behavior when we do not have a retry policy. 
Thanks
